pybamm/spatial_methods/finite_volume.py

```diff
@@ -1121,10 +1140,138 @@ def arithmetic_mean(array):
 
             return pybamm.Matrix(matrix) @ array
 
+        def harmonic_mean(array):
+            """
+            Calculate the harmonic mean of an array using matrix multiplication.
+            The harmonic mean is computed as
+
+            .. math::
+                D_{eff} = \\frac{D_1  D_2}{\\beta D_2 + (1 - \\beta) D_1},
+
+            where
+
+            .. math::
+                \\beta = \\frac{\\Delta x_1}{\\Delta x_2 + \\Delta x_1}
+
+            accounts for the difference in the control volume widths. This is the
+            definiton from [1].
```

Comment:
is this the same definition as the Recktenwald paper?

Reply:
yeah, just in slightly different notation. Could remove this ref if you think it's cleaner to just provide one

Reply:
just saying it's the same but with different notation would be fine (you can add that commit with [ci skip] if you don't want to wait for the tests)